File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if head==None:
            return None
        f=s=head
        for i in range(k-1):
            f=f.next
        t=f
        while(t.next!=None):
            s=s.next
            t=t.next
        f.val,s.val=s.val,f.val
        return head
        

File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapNodes(ListNode* head, int k) {
        int count=0;
        ListNode* curr=head;
        while(curr!=nullptr)
        {
            count++;
            curr=curr->next;
        }
        ListNode* one=nullptr,*two=nullptr;
        curr=head;
        int c=0;
        while(curr!=nullptr)
        {
            c++;
            if(c==k)one=curr;
            if(c==count-k+1)two=curr;
            curr=curr->next;
        }
        int temp=one->val;
        one->val=two->val;
        two->val=temp;
        return head;
    }
};